src/otx/engine/hpo/hpo_api.py

```diff
@@ -177,7 +178,7 @@ def _get_default_search_space(self) -> dict[str, Any]:
         if isinstance(self._engine.optimizer, list):
             for i, optimizer in enumerate(self._engine.optimizer):
                 search_space[f"optimizer.{i}.keywords.lr"] = self._make_lr_search_space(optimizer)
-        elif isinstance(self._engine.optimizer, OptimizerCallable):
+        elif self._engine.optimizer is not None:
```

Comment:
Why this change is needed? Is there any case the optimizer isn't `OptimizerCallable`?

Reply:
I'm not sure, @eunwoosh Could you explain this?

Reply:
No, but `OptimizerCallable` can't be used for `isinstance` because it just a `Callable` instance.